File: Backend/src/main/java/com/project/consphere/service/LikeService.java
package com.project.consphere.service;

import com.project.consphere.dto.LikeResponse;

public interface LikeService {

    LikeResponse likePost(Long postId);

    LikeResponse unlikePost(Long postId);

    LikeResponse getLikeStatus(Long postId);
}

File: frontend/src/services/likeService.js
import api from './api';

export const likeService = {
  likePost: async (postId) => {
    const response = await api.post(`/likes/post/${postId}`);
    return response.data;
  },

  unlikePost: async (postId) => {
    const response = await api.delete(`/likes/post/${postId}`);
    return response.data;
  },

  getLikeStatus: async (postId) => {
    const response = await api.get(`/likes/post/${postId}`);
    return response.data;
  }
};
